tests/xrt/100_ert_ncu/xrtxx-ip.cpp

```diff
@@ -195,6 +199,33 @@ struct job_type
     }
   }
 
+  void
+  run_intr(std::chrono::milliseconds& timeout)
+  {
+    auto interrupt = ip.create_interrupt_notify();
+
+    while (1) {
+      ip.write_register(0, AP_START);
+      ++runs;
+      auto ret = interrupt.wait(timeout);
+      if (ret == std::cv_status::timeout)
+        throw std::runtime_error("Timeout when waiting for CU interrupt");
```

Comment:
Better not throw for user specified timeout.  How about looping with same timeout and counting how many loops you make similar to run_poll()?   Either allow test to hang if looping never completes (that would be an error), or put in some smarts that make it throw after after say 1 second?

Reply:
updated